src/MOL_discretization.jl

```diff
@@ -1,22 +1,220 @@
+#using Reduce
+
+# Method of lines discretization scheme
 struct MOLFiniteDifference{T} <: DiffEqBase.AbstractDiscretization
   dxs::T
   order::Int
 end
 MOLFiniteDifference(args...;order=2) = MOLFiniteDifference(args,order)
 
+# Get boundary conditions from an array
+function get_bcs(bcs,tdomain,domain)
+    lhs_deriv_depvars_bcs = Dict()
+    n = size(bcs,1)
+    for i = 1:n
+        var = bcs[i].lhs.op
+        if var isa Variable
+            if !haskey(lhs_deriv_depvars_bcs,var)
+                lhs_deriv_depvars_bcs[var] = Array{Expr}(undef,3)
+            end
+            if isequal(bcs[i].lhs.args[1],tdomain.lower) # u(t=0,x) 
+                lhs_deriv_depvars_bcs[var][1] = Expr(bcs[i].rhs)
+            elseif isequal(bcs[i].lhs.args[2],domain.lower) # u(t,x=x_init)
+                lhs_deriv_depvars_bcs[var][2] = :(var=$(bcs[i].rhs.value))
+            elseif isequal(bcs[i].lhs.args[2],domain.upper) # u(t,x=x_final)
+                lhs_deriv_depvars_bcs[var][3] = :(var=$(bcs[i].rhs.value))
+            end
+        end
+    end
+    return lhs_deriv_depvars_bcs
+end

```

Comment:
This should be building a Q operator?

Reply:
It is a possibility, we should take into account that the BCs should be evaluated w.r.t. t at each time step within "f(du,u,p,t)".

Reply:
Yes, Q is lazy so you'd just create the type on the fly before evaluation.

Reply:
Ok. I assume this is planned for an upcoming PR specializing in BCs, is that right?

Reply:
That should already work

Reply:
Note: the function get_bcs works, the Q operator is created below in the code:
```
### Get boundary conditions ################################
    # TODO: generalize to N equations
    lhs_deriv_depvars_bcs = get_bcs(pdesys.bcs,tdomain,domain[1])
    t = 0.0
    interior = domain[1].lower+dx[1]:dx[1]:domain[1].upper-dx[1]
    u0 = Array{Float64}(undef,length(interior),length(discretization))
    Q = Array{RobinBC}(undef,length(discretization))
    
    i = 1
    for var in keys(discretization)
        bcs = lhs_deriv_depvars_bcs[var]
        g = eval(:((x,t) -> $(bcs[1])))
        u0[:,i] = @eval $g.($interior,$t)
        u_x0 = eval(bcs[2])
        u_x1 = eval(bcs[3])
        Q[i] = DirichletBC(u_x0,u_x1)
        i = i+1
    end
```